### PR TITLE
Create initial repo

### PR DESCRIPTION
**Pull request by LizaShak**
# PR for issue

## What is being addressed

Create initial repository to extend AzureTRE resources and have CI/CD pipelines in place

## How is this addressed

- Devcontainer definition
- CICD github actions
- Directories structure
- AzureTRE Makefile commands included

### CODE REVIEW

File: .devcontainer/scripts/install-azure-tre-oss.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+sed -i 's,\./ui,${AZURETRE_HOME}/ui,g' "$oss_home"/devops/scripts/build_deploy_ui.sh
+# shellcheck disable=SC2016
+sed -i 's,\./devops,${AZURETRE_HOME}/devops,g' "$oss_home"/devops/scripts/build_deploy_ui.sh
+
+sed -i 's,masked_resource.properties,# masked_resource.properties,g' "$oss_home"/api_app/api/routes/resource_helpers.py
+echo "__version__ = \"0.3.23.2\"" > "$oss_home"/api_app/_version.py
+
+
+# waiting for: https://github.com/microsoft/AzureTRE/pull/2248
+sed -i 's/lstrip('\''api:\/\/'\'')/replace('\''api:\/\/'\'','\'''\'')/g' "$oss_home"/api_app/api/routes/api.py
```
Comment (tamirkamara) [inline .devcontainer/scripts/install-azure-tre-oss.sh:43]: lets delete all of this

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  "build": {
+    "context": "..",
+    "dockerfile": "Dockerfile",
+    "args": {
+      // To ensure that the group ID for the docker group in the container
+      // matches the group ID on the host, add this to your .bash_profile on the host
+      // 		export DOCKER_GROUP_ID=$(getent group docker | awk -F ":" '{ print $3 }')
+      "DOCKER_GROUP_ID": "${localEnv:DOCKER_GROUP_ID}",
+      "INTERACTIVE": "true",
+      "OSS_VERSION": "c28403959579e82b93cf899b93b9d3aeb883d555"
```
Comment (tamirkamara) [inline .devcontainer/devcontainer.json:14]:
```suggestion
      "OSS_VERSION": "v0.4"
```

**Review by tamirkamara — CHANGES_REQUESTED** [review]
@LizaShak Are the copied files insync with the ones on the parent repo?

**Comment by stuartleeks** [comment]
Having the `devops/.env.sample` in this repo makes sense (i.e. I think that's the right location for the `.env` file in this setup), but when I run `make auth` after creating `devops/.env` I still get the `»»» ⚠️ Your devops environment has not been setup! 😥 Please create a ./devops/.env file` error message.

After a quick look at the `check_dependencies.sh` script, I think it is looking for the `.env` file relative to the script rather than based on the current directory, so I think we will need to update the script in the OSS repo for this

**Comment by tamirkamara** [comment]
> Having the `devops/.env.sample` in this repo makes sense (i.e. I think that's the right location for the `.env` file in this setup), but when I run `make auth` after creating `devops/.env` I still get the `»»» ⚠️ Your devops environment has not been setup! 😥 Please create a ./devops/.env file` error message.
> 
> After a quick look at the `check_dependencies.sh` script, I think it is looking for the `.env` file relative to the script rather than based on the current directory, so I think we will need to update the script in the OSS repo for this
I knew that something will break with that recent .env changes...

**Comment by stuartleeks** [comment]
@LizaShak - I'm running through trying to deploy an environment from this PR code, do you want me to PR the script changes that I'm making back to AzureTRE?

File: .devcontainer/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+RUN export PORTER_MIRROR=${PORTER_MIRROR} \
+    PORTER_PERMALINK=${PORTER_PERMALINK} \
+    PORTER_PKG_PERMALINK=${PORTER_PKG_PERMALINK} \
+    PORTER_HOME=${PORTER_HOME} \
+    && /tmp/porter.sh
+
+ENV PATH ${PORTER_HOME}:$PATH
+
+# Install azure-cli
+ARG AZURE_CLI_VERSION=2.36.0-1~buster
```
Comment (stuartleeks) [inline .devcontainer/Dockerfile:73]: The auth scripts in AzureTRE have recently been updated, and AFAIK now require 2.37 or later of `az`

**Comment by LizaShak** [comment]
> @LizaShak - I'm running through trying to deploy an environment from this PR code, do you want me to PR the script changes that I'm making back to AzureTRE?

sure, sounds great

File: .devcontainer/Dockerfile (hunk 3 above)
Comment (LizaShak) [reply .devcontainer/Dockerfile]: updated

**Review by stuartleeks — APPROVED** [review]
LGTM

**Review by tamirkamara — APPROVED** [review]

**Pull request merged**